blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineRunImpl.java

```diff
@@ -107,6 +108,22 @@ public BlueRunState getStateObj() {
         } catch (InterruptedException | TimeoutException e) {
             logger.error("Error getting StateObject from execution context: "+e.getMessage(), e);
         }
+
+        // TODO: Probably move this elsewhere - maybe into PipelineNodeContainerImpl?
+        boolean isQueued = false;
+        boolean isRunning = false;
+        for (BluePipelineNode n : getNodes()) {
+            if (n.getStateObj().equals(BlueRunState.QUEUED)) {
+                isQueued = true;
+            } else if (n.getStateObj().equals(BlueRunState.RUNNING)) {
+                isRunning = true;
+            }
+        }
+        if (isQueued && !isRunning) {
+            return BlueRunState.QUEUED;
+
+        }
+
         return super.getStateObj();
     }
```

Comment:
Make this whole method assign a single variable `state` instead of returning immediately (including `super.getStateObj()`) then `return state;` on last line. Set a breakpoint there and you should find the null pretty easily.